roachtest: activerecord failed

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5148992&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5148992&tab=artifacts#/activerecord) on master @ [bc1ee7c7c276984fce8ff5ba4fcfcdff335dde50](https://github.com/cockroachdb/cockroach/commits/bc1ee7c7c276984fce8ff5ba4fcfcdff335dde50):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1584-gbc1ee7c7c2
		Tests run against activerecord 6.1.5
		6664 Total Tests Run
		6662 tests passed
		2 tests failed
		18 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: InsertAllTest#test_insert_all_and_upsert_all_with_expression_index (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #81087 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5152767&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5152767&tab=artifacts#/activerecord) on master @ [9a179ea1aa8d6723fb12a988f2212ac8493e5dfc](https://github.com/cockroachdb/cockroach/commits/9a179ea1aa8d6723fb12a988f2212ac8493e5dfc):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1588-g9a179ea1aa
		Tests run against activerecord 6.1.5
		6664 Total Tests Run
		6662 tests passed
		2 tests failed
		18 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: InsertAllTest#test_insert_all_and_upsert_all_with_expression_index (unexpected)
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #81087 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5157842&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5157842&tab=artifacts#/activerecord) on master @ [16c484aa84d3718bfc82557f8e935ab78e6753b6](https://github.com/cockroachdb/cockroach/commits/16c484aa84d3718bfc82557f8e935ab78e6753b6):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1601-g16c484aa84
		Tests run against activerecord 6.1.6
		6666 Total Tests Run
		6665 tests passed
		1 test failed
		19 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #81087 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5163863&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5163863&tab=artifacts#/activerecord) on master @ [bc5f5b7159ed2b7dbffa94bb2a590fdddb642b9b](https://github.com/cockroachdb/cockroach/commits/bc5f5b7159ed2b7dbffa94bb2a590fdddb642b9b):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1607-gbc5f5b7159
		Tests run against activerecord 6.1.6
		6666 Total Tests Run
		6665 tests passed
		1 test failed
		19 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #81087 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5176233&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5176233&tab=artifacts#/activerecord) on master @ [2c264bd12c15b8811553adc9d90f9230065f3ff3](https://github.com/cockroachdb/cockroach/commits/2c264bd12c15b8811553adc9d90f9230065f3ff3):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1649-g2c264bd12c
		Tests run against activerecord 6.1.6
		6666 Total Tests Run
		6665 tests passed
		1 test failed
		19 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5190042&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5190042&tab=artifacts#/activerecord) on master @ [a35b848edb3f5562199b0f56bc27a6c576073d5d](https://github.com/cockroachdb/cockroach/commits/a35b848edb3f5562199b0f56bc27a6c576073d5d):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1716-ga35b848edb
		Tests run against activerecord 6.1.6
		6666 Total Tests Run
		6665 tests passed
		1 test failed
		19 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5202348&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5202348&tab=artifacts#/activerecord) on master @ [ce66acdbba801de88f1dd645eaedeb3834f23dbd](https://github.com/cockroachdb/cockroach/commits/ce66acdbba801de88f1dd645eaedeb3834f23dbd):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/activerecord/run_1
	orm_helpers.go:193,orm_helpers.go:119,activerecord.go:251,test_runner.go:876: 
		Tests run on Cockroach v22.1.0-alpha.4-1757-gce66acdbba
		Tests run against activerecord 6.1.6
		6666 Total Tests Run
		6665 tests passed
		1 test failed
		19 tests skipped
		5 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: ActiveRecord::Migration::IndexTest#test_rename_index_with_symbol (unexpected)
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList22_1) is available in the artifacts' activerecord log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81088 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #80825 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1.0]
- #79491 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/rename_index
+++ b/pkg/sql/logictest/testdata/logic_test/rename_index
@@ -169,3 +169,19 @@ SELECT DISTINCT index_name FROM [SHOW INDEXES FROM users]
 pk
 ufo
 bar
+
+# Regression test for #81211
+statement ok
+CREATE TABLE t1(a int);
+
+statement ok
+BEGIN;
+
+statement ok
+CREATE INDEX i1 ON t1(a);
+
+statement ok
+ALTER INDEX i1 RENAME TO i2;
+
+statement ok
+COMMIT;

--- a/pkg/sql/resolver.go
+++ b/pkg/sql/resolver.go
@@ -662,7 +662,7 @@ func expandMutableIndexName(
 func expandIndexName(
 	ctx context.Context,
 	txn *kv.Txn,
-	sc resolver.SchemaResolver,
+	p *planner,
 	codec keys.SQLCodec,
 	index *tree.TableIndexName,
 	requireTable bool,
@@ -670,7 +670,7 @@ func expandIndexName(
 	tn = &index.Table
 	if tn.Object() != "" {
 		// The index and its table prefix must exist already. Resolve the table.
-		_, desc, err = resolver.ResolveMutableExistingTableObject(ctx, sc, tn, requireTable, tree.ResolveRequireTableOrViewDesc)
+		_, desc, err = resolver.ResolveMutableExistingTableObject(ctx, p, tn, requireTable, tree.ResolveRequireTableOrViewDesc)
 		if err != nil {
 			return nil, nil, err
 		}
@@ -681,7 +681,7 @@ func expandIndexName(
 		return tn, desc, nil
 	}
 
-	found, resolvedPrefix, tbl, _, err := resolver.ResolveIndex(ctx, sc, index, txn, codec, requireTable, false /*requireActiveIndex*/)
+	found, resolvedPrefix, tbl, _, err := resolver.ResolveIndex(ctx, p, index, txn, codec, requireTable, false /*requireActiveIndex*/)
 	if err != nil {
 		return nil, nil, err
 	}
@@ -698,7 +698,10 @@ func expandIndexName(
 	// Memoize the table name that was found. tn is a reference to the table name
 	// stored in index.Table.
 	*tn = tableName
-	tblMutable := tbl.NewBuilder().BuildExistingMutable().(*tabledesc.Mutable)
+	tblMutable, err := p.Descriptors().GetMutableTableVersionByID(ctx, tbl.GetID(), p.Txn())
+	if err != nil {
+		return nil, nil, err
+	}
 	return &tableName, tblMutable, nil
 }
 

